packages/intrepid2/unit-test/MonolithicExecutable/BasisValuesTests.cpp

```diff
@@ -0,0 +1,325 @@
+    }
+    else
+    {
+      std::vector<ViewType<PointScalar>> pointComponents {points};
+      tensorPoints = TensorPoints<PointScalar>(pointComponents);
+      Data<WeightScalar> weightData(weights);
+      std::vector<Data<WeightScalar>> weightComponents {weightData};
+      tensorWeights = TensorData<WeightScalar>(weightComponents);
+    }
+    
+    out << "Points being tested:\n";
+    for (int pointOrdinal=0; pointOrdinal<numRefPoints; pointOrdinal++)
+    {
+      out << pointOrdinal << ": " << "(";
+      for (int d=0; d<spaceDim; d++)
+      {
+        out << points(pointOrdinal,d);
+        if (d < spaceDim-1) out << ",";
+      }
+      out << ")\n";
+    }
+    
+    printFunctor2(tensorPoints, out, "tensorPoints");
+    
+    testFloatingEquality2(points,tensorPoints,  relTol, absTol, out, success, "points", "tensorPoints");
+        
+    for (const auto &op : opsToTest)
+    {
+      auto basisValuesView = basis.allocateOutputView(numRefPoints, op);
+      auto basisValues     = basis.allocateBasisValues(tensorPoints, op);
+      
+      basis.getValues(basisValuesView, points, op);
+      basis.getValues(basisValues, tensorPoints, op);
+      
+      out << "Comparing getValues() results for " << EOperatorToString(op) << std::endl;
+      
+      TEST_EQUALITY(basisValues.rank(), basisValuesView.rank());
+
+      if (basisValues.rank() == basisValuesView.rank())
+      {
+        if (basisValuesView.rank() == 2)
+        {
+          bool localSuccess = true;
+          testFloatingEquality2(basisValuesView, basisValues, relTol, absTol, out, localSuccess, "DynRankView path", "BasisValues path");
+          success = success && localSuccess;
+          
+          if (!localSuccess)
+          {
+            printFunctor2(basisValues,     out, "basisValues");
+            printFunctor2(basisValuesView, out, "basisValuesView");
+          }
+        }
+        else if (basisValuesView.rank() == 3)
+        {
+          bool localSuccess = true;
+          testFloatingEquality3(basisValuesView, basisValues, relTol, absTol, out, localSuccess, "DynRankView path", "BasisValues path");
+          success = success && localSuccess;
+          
+          if (!localSuccess)
+          {
+            printFunctor3(basisValues,     out, "basisValues");
+            printFunctor3(basisValuesView, out, "basisValuesView");
+          }
+        }
+        else
+        {
+          INTREPID2_TEST_FOR_EXCEPTION(true, std::logic_error, "Unexpected rank encountered for basisValuesView");
+        }
+      }
+      else
+      {
+        out << "FAILURE: basisValues.rank() does not match basisValuesView.rank().\n";
+        if (basisValues.rank() == 2)
+        {
+          printFunctor2(basisValues, out, "basisValues");
+        }
+        else if (basisValues.rank() == 3)
+        {
+          printFunctor3(basisValues, out, "basisValues");
+        }
+      }
+    }
+  }
+
+  TEUCHOS_UNIT_TEST( BasisValues, HierarchicalHGRAD_LINE )
```

Comment:
Yes, it would.  There aren't yet HDIV, HCURL hierarchical bases for those, so the best we can do for those is test the nodal bases.  I've now added those tests here, as well as TRI/TET tests against HGRAD hierarchical.

Reply:
Thanks!